pkg/kubectl/cmd/label.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+func (f *Factory) NewCmdLabel(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "label [--overwrite] <resource> <name> <key-1>=<val-1> ... <key-n>=<val-n> [--resource-version=<version>]",
```

Comment:
What's the most common action?  Adding/overwriting, then clearing?  If adding is sufficiently more prevalent that clearing, it's an argument to make the common action use the arguments.


> On Feb 7, 2015, at 1:46 PM, Jeff Lowdermilk <notifications@github.com> wrote:
> 
> In pkg/kubectl/cmd/label.go:
> 
> > +import (
> > +	"fmt"
> > +	"io"
> > +	"strings"
> > +
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
> > +	"github.com/spf13/cobra"
> > +)
> > +
> > +func (f *Factory) NewCmdLabel(out io.Writer) *cobra.Command {
> > +	cmd := &cobra.Command{
> > +		Use:   "label [--overwrite] <resource> <name> <key-1>=<val-1> ... <key-n>=<val-n> [--resource-version=<version>]",
> A thought for using this to clear labels as well; make the labels to apply a flag and add a flag to clear labels. E.g.
> 
> label [--overwrite] <resource> <name> --labels=<key-1>=<val-1>... --clear-labels=<key-2>,<key-3>
> —
> Reply to this email directly or view it on GitHub.
>